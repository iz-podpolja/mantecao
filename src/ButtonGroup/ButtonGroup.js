import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import ButtonGroupItem from './_ButtonGroupItem'

const StyledDiv = styled.div`
  display: flex;
  flex-wrap: wrap;
`

class ButtonGroup extends Component {
  render() {
    const { options, optionValues, disabled, selected } = this.props
    return (
      <StyledDiv>
        {options.map((option, index) => {
          return (
            <ButtonGroupItem
              key={index}
              first={index === 0}
              last={index === options.length - 1}
              value={option}
              text={optionValues[index]}
              selected={selected.indexOf(option) > -1}
              onSelect={this._toggleOption}
              disabled={disabled}
            />
          )
        })}
      </StyledDiv>
    )
  }
  _toggleOption = option => {
    const { onChange, selected } = this.props
    let newArray = selected.slice()
    let index = selected.indexOf(option)
    if (index > -1) {
      newArray.splice(index, 1)
    } else {
      newArray.push(option)
    }
    onChange(newArray)
  }
}

ButtonGroup.propTypes = {
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  optionValues: PropTypes.array.isRequired,
  options: PropTypes.array.isRequired,
  selected: PropTypes.array.isRequired
}

export default ButtonGroup
