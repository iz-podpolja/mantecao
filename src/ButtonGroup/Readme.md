Example:

    <ButtonGroup
      options={[1, 2, 3, 4, 5]}
      optionValues={['a', 'b', 'c', 'd', 'e']}
      selected={[1,4,5]}
      onChange={(values) => console.log(values)}
    />
