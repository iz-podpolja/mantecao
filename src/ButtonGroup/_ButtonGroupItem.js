import React from 'react'
import PropTypes from 'prop-types'
import { darken } from 'polished'

import { colors } from '../variables'
import Button from '../Button'

const ButtonGroupItem = ({
  text,
  onSelect,
  disabled,
  first,
  last,
  selected,
  value
}) => {
  const style = {
    borderTopLeftRadius: first ? '4px' : '0px',
    borderBottomLeftRadius: first ? '4px' : '0px',
    borderTopRightRadius: last ? '4px' : '0px',
    borderBottomRightRadius: last ? '4px' : '0px',
    boxShadow: 'inset 0 1px rgba(0,0,0,.05)',
    padding: '.5em 1.5em',
    borderLeft: `1px solid ${selected
      ? darken(0.1, colors.blueBayoux)
      : darken(0.1, colors.silver)}`
  }

  return (
    <Button
      onClick={() => onSelect(value)}
      color={selected ? colors.blueBayoux : colors.silver}
      disabled={disabled}
      style={style}
    >
      {text}
    </Button>
  )
}

const { bool, func, number, string, oneOfType } = PropTypes
ButtonGroupItem.propTypes = {
  disabled: bool,
  first: bool.isRequired,
  last: bool.isRequired,
  onSelect: func.isRequired,
  selected: bool.isRequired,
  text: oneOfType([string, number]).isRequired,
  value: oneOfType([string, number, func]).isRequired
}

export default ButtonGroupItem
