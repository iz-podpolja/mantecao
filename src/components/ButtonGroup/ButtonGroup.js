import React, {PropTypes, Component} from 'react'

import ButtonGroupItem from './_ButtonGroupItem'

class ButtonGroup extends Component {
  render () {
    const {options, optionValues, disabled, selected} = this.props
    return (
      <div style={{display: 'flex', flexWrap: 'wrap'}}>
        {
          options.map((option, index) => {
            return (
              <ButtonGroupItem
                key={index}
                first={index === 0}
                last={index === (options.length - 1)}
                value={option}
                text={optionValues[index]}
                selected={selected.indexOf(option) > -1}
                onSelect={this._toggleOption}
                disabled={disabled ? disabled.indexOf(option) > -1 : undefined}
              />
            )
          })
        }
      </div>
    )
  }
  _toggleOption = (option) => {
    const {onChange, selected} = this.props
    let newArray = selected.slice()
    let index = selected.indexOf(option)
    if (index > -1) {
      newArray.splice(index, 1)
    } else {
      newArray.push(option)
    }
    onChange(newArray)
  }
}

ButtonGroup.propTypes = {
  disabled: PropTypes.array,
  onChange: PropTypes.func,
  optionValues: PropTypes.array.isRequired,
  options: PropTypes.array.isRequired,
  selected: PropTypes.array.isRequired
}

export default ButtonGroup
