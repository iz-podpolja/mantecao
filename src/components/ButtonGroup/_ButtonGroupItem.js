import React, { PropTypes, Component } from 'react'
import { Button } from '../Button'

import { colors } from '../../variables'

const itemStyle = {
  display: 'inline-block',
  fontSize: '1.4rem',
  padding: '0.3rem 2rem',
  borderRadius: '0px',
  border: '1px solid',
  boxShadow: 'inset 0 1px rgba(0,0,0,.05)'
}
const firstItemStyle = {
  borderTopLeftRadius: '4px',
  borderBottomLeftRadius: '4px'
}
const middleItemStyle = {
  borderLeft: 'none'
}
const lastItemStyle = {
  borderTopRightRadius: '4px',
  borderBottomRightRadius: '4px'
}
const selectedItemStyle = {
  backgroundColor: colors.pacificBlue,
  borderColor: colors.deepCerulean
}
const unselectedItemStyle = {
  borderColor: colors.silver
}

class ButtonGroupItem extends Component {
  render() {
    const {
      text,
      value,
      onSelect,
      first,
      last,
      selected,
      disabled
    } = this.props
    const baseStyle = Object.assign(
      {},
      itemStyle,
      first ? firstItemStyle : middleItemStyle,
      last ? lastItemStyle : {},
      selected ? selectedItemStyle : unselectedItemStyle
    )
    return (
      <Button
        style={baseStyle}
        onClick={() => onSelect(value)}
        disabled={disabled}
      >
        {text}
      </Button>
    )
  }
}

const { bool, func, number, string, oneOfType } = PropTypes
ButtonGroupItem.propTypes = {
  disabled: bool,
  first: bool,
  last: bool,
  onSelect: func,
  selected: bool,
  text: oneOfType([string, number]),
  value: oneOfType([string, number, func])
}

export default ButtonGroupItem
