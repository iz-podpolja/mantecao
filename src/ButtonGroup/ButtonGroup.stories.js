import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import ButtonGroup from './ButtonGroup'

storiesOf('ButtonGroup', module).add('default', () =>
  <div>
    <ButtonGroup
      options={[1, 2, 3, 4, 5]}
      optionValues={['a', 'b', 'c', 'd', 'e']}
      selected={[1, 4, 5]}
      onChange={action('onChange')}
    />
  </div>
)
