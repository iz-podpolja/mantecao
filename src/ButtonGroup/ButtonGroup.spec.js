import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ButtonGroup from './ButtonGroup'
import ButtonGroupItem from './_ButtonGroupItem'

describe('ButtonGroup Component', () => {
  it('renders the correct selected and unselected things', () => {
    const tree = renderer
      .create(
        <ButtonGroup
          options={[1, 2, 3, 4, 5]}
          optionValues={['a', 'b', 'c', 'd', 'e']}
          selected={[1, 4, 5]}
          onChange={() => false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders properly with all selected', () => {
    const tree = renderer
      .create(
        <ButtonGroup
          options={[1, 2, 3, 4, 5]}
          optionValues={['a', 'b', 'c', 'd', 'e']}
          selected={[1, 2, 3, 4, 5]}
          onChange={() => false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders properly with none selected', () => {
    const tree = renderer
      .create(
        <ButtonGroup
          options={[1, 2, 3, 4, 5]}
          optionValues={['a', 'b', 'c', 'd', 'e']}
          selected={[]}
          onChange={() => false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders properly disabled', () => {
    const tree = renderer
      .create(
        <ButtonGroup
          options={[1, 2, 3, 4, 5]}
          optionValues={['a', 'b', 'c', 'd', 'e']}
          selected={[1, 4, 5]}
          onChange={() => false}
          disabled
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('fires onChange function properly', () => {
    const spy = jest.fn()
    const component = shallow(
      <ButtonGroup
        options={[1, 2, 3, 4, 5]}
        optionValues={['a', 'b', 'c', 'd', 'e']}
        selected={[1, 4, 5]}
        onChange={spy}
        disabled
      />
    )
    const firstItem = component.find(ButtonGroupItem).first()
    firstItem.props().onSelect(firstItem.props().value)
    expect(spy.mock.calls[0][0]).toEqual([4, 5])
  })
})
