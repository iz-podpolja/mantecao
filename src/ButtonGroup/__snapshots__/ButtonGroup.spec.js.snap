// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component renders properly disabled 1`] = `
<div
  className="sc-bwzfXH icOaIC"
>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    a
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    b
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    c
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    d
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "4px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "4px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    e
  </button>
</div>
`;

exports[`ButtonGroup Component renders properly with all selected 1`] = `
<div
  className="sc-bwzfXH icOaIC"
>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    a
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    b
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    c
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    d
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "4px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "4px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    e
  </button>
</div>
`;

exports[`ButtonGroup Component renders properly with none selected 1`] = `
<div
  className="sc-bwzfXH icOaIC"
>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    a
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    b
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    c
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    d
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "4px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "4px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    e
  </button>
</div>
`;

exports[`ButtonGroup Component renders the correct selected and unselected things 1`] = `
<div
  className="sc-bwzfXH icOaIC"
>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    a
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    b
  </button>
  <button
    className="sc-bdVaJa dNukzi"
    color="#CCCCCC"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #b3b3b3",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    c
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "0px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    d
  </button>
  <button
    className="sc-bdVaJa dasgOI"
    color="#556285"
    disabled={undefined}
    onClick={[Function]}
    style={
      Object {
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "4px",
        "borderLeft": "1px solid #414b66",
        "borderTopLeftRadius": "0px",
        "borderTopRightRadius": "4px",
        "boxShadow": "inset 0 1px rgba(0,0,0,.05)",
        "padding": ".5em 1.5em",
      }
    }
  >
    e
  </button>
</div>
`;
